src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -415,9 +422,12 @@ public IEnumerable<FilterLog> FindLogs(LogFilter filter, CancellationToken cance
             return _logFinder.FindLogs(filter, cancellationToken);
         }
 
-        private string? ConstructError(TransactionResult txResult, string? tracerError, long gasLimit)
+        private static string? ConstructError(TransactionResult txResult, string? tracerError, long gasLimit)
         {
-            if (txResult.Success) return tracerError;
+            if (txResult.Success)
+            {
+                return tracerError is not null ? $"err: {tracerError} (supplied gas {gasLimit})" : null;
+            }
             return txResult.Error is not null ? $"err: {txResult.Error} (supplied gas {gasLimit})" : null;
```

Comment:
Merge conditionals to avoid duplicating the error message:

```suggestion
            var error = txResult switch
            {
                { Success: true } when tracerError is not null => tracerError,
                { Success: false, Error: not null } => txResult.Error,
                _ => null
            };

            return error is null ? null : $"err: {error} (supplied gas {gasLimit})";
```